src/index.js

```diff
@@ -402,13 +403,21 @@ const draw = (data, container, imagesPath, labelLimit = 8, rankDir = 'LR') => {
       );
     }
 
-    // this will allow the labels to be turned off if there are too many nodes and edge labels overlap
-    g.nodeCount() < labelLimit && createControlText(index, controlText);
-    g.nodeCount() < labelLimit && createOwnText(index, shareText);
-    g.nodeCount() < labelLimit &&
-      !('shareholding' in interests) &&
-      !('votingRights' in interests) &&
-      createUnknownText(index, element);
+    const limitLabels = (createLabel) => g.nodeCount() < labelLimit && createLabel;
+
+    // this creates the edge labels, and will allow the labels to be turned off if the node count exceeds the labelLimit
+    limitLabels(createControlText(index, controlText));
+    limitLabels(createOwnText(index, shareText));
+
+    // The unknown interest labels are drawn when the interest type is set to 'unknownInterest' or the data are missing
+    // No label is displayed if the interestType is within the interestType codelist but is not shareholding or votingRights
+    if (
```

Comment:
My understanding of how this is dealing with multiple interests is that where *any* of the interests are shareholding or voting rights, this will be displayed, else if *any* are in the codelist then nothing is displayed, else display the unknown interest type. I think this makes sense, but just wanted to check that this is correct, and the logic we want (cc @kd-ods)

Reply:
Yes - And it looks like it picks up the 'unknownInterest' type too and displays the relevant label. So its all good @lgs85 @Lathrisk